grpc-kotlin/src/test/proto/com/linecorp/armeria/grpc/testing/test.proto

```diff
@@ -0,0 +1,105 @@
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// An integration test service that covers all the method signature permutations
+// of unary/streaming requests/responses.
+syntax = "proto3";
+
+import "com/linecorp/armeria/grpc/testing/empty.proto";
+import "com/linecorp/armeria/grpc/testing/messages.proto";
+import "google/api/annotations.proto";
+
+package armeria.grpc.testing;
+
+option java_package = "com.linecorp.armeria.grpc.testing";
+
+// A simple service to test the various types of RPCs and experiment with
+// performance with various types of payload.
+service TestService {
+  // One empty request followed by one empty response.
+  rpc EmptyCall(armeria.grpc.testing.Empty) returns (armeria.grpc.testing.Empty);
+
+  // One request followed by one response.
+  rpc UnaryCall(SimpleRequest) returns (SimpleResponse);
+
+  // Another method with one request followed by one response.
+  rpc UnaryCall2(SimpleRequest) returns (SimpleResponse) {
+    // Comment about an option
+    option (google.api.http) = {
+      post: "/v1/unary-call-2"
+      body: "*"
+    };
+  }
+
+  // One request followed by a sequence of responses (streamed download).
+  // The server returns the payload with client desired type and sizes.
+  rpc StreamingOutputCall(StreamingOutputCallRequest)
+      returns (stream StreamingOutputCallResponse);
+
+  // A sequence of requests followed by one response (streamed upload).
+  // The server returns the aggregated size of client payload as the result.
+  rpc StreamingInputCall(stream StreamingInputCallRequest)
+      returns (StreamingInputCallResponse);
+
+  // A sequence of requests with each request served by the server immediately.
+  // As one request could lead to multiple responses, this interface
+  // demonstrates the idea of full duplexing.
+  rpc FullDuplexCall(stream StreamingOutputCallRequest)
+      returns (stream StreamingOutputCallResponse);
+
+  // A sequence of requests followed by a sequence of responses.
+  // The server buffers all the client requests and then serves them in order. A
+  // stream of responses are returned to the client when the server starts with
+  // first request.
+  rpc HalfDuplexCall(stream StreamingOutputCallRequest)
+      returns (stream StreamingOutputCallResponse);
+
+  // The test server will not implement this method. It will be used
+  // to test the behavior when clients call unimplemented methods.
+  rpc UnimplementedCall(armeria.grpc.testing.Empty) returns (armeria.grpc.testing.Empty);
+
+  // This method's parameter message contains all different types of parameters
+  // as well as the response type contains all different types of parameters.
+  // Can be used to check any kind of serialization issues.
+  rpc UnaryCallWithAllDifferentParameterTypes(ExtendedTestMessage) returns (ExtendedTestMessage);
```

Comment:
```suggestion
```

Reply:
Done in [c5aa9ab](https://github.com/line/armeria/pull/4724/commits/c5aa9ab3a1e5d1d4bfa41274d85c350ec4da290d)!